### PR TITLE
shared styles arrays

### LINKED ISSUE #355: SharedStyles should be an array property
`document.getSharedLayerStyles()` returns an array but you can't modify it so you can't remove a shared style for example.

It being an array property would allow that

### PR DESCRIPTION
**Pull request by mathieudutour**
fix #355

### CODE REVIEW

**Comment by bohemian-coding-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This pull request may need a CHANGELOG entry.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  `Source/dom/WrappedObject.js` changed, but not:
  - 🧪 its tests

That's OK as long as you're refactoring.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  `Source/dom/layers/Group.js` changed, but not:
  - 📚 its docs
- 🧪 its tests

That's OK as long as you're refactoring.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  `Source/dom/layers/Layer.js` changed, but not:
  - 📚 its docs
- 🧪 its tests

That's OK as long as you're refactoring.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  `Source/dom/models/SharedStyle.js` changed, but not:
  - 🧪 its tests

That's OK as long as you're refactoring.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  `Source/dom/style/Style.js` changed, but not:
  - 📚 its docs
- 🧪 its tests

That's OK as long as you're refactoring.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Source/dom/models/SharedStyle.js [hunk 1]
```diff
@@ -15,13 +15,16 @@ export class SharedStyle extends WrappedObject {
   constructor(master = {}) {
     if (!master.sketchObject) {
       throw new Error(
-        `Cannot create a SharedStyle directly, use SharedStyle.fromStyle({ name, style, document }) instead.`
+        `Cannot create a SharedStyle directly, use \`document.sharedLayersStyles.push({ name, style })\` (or \`document.sharedTextStyles\`) instead.`
```
Comment (bomberstudios) [inline Source/dom/models/SharedStyle.js]: Typo:
```suggestion
        `Cannot create a SharedStyle directly, use \`document.sharedLayerStyles.push({ name, style })\` (or \`document.sharedTextStyles\`) instead.`
```

File: Source/dom/models/SharedStyle.js [hunk 2]
```diff
@@ -15,13 +15,16 @@ export class SharedStyle extends WrappedObject {
...
       )
     }
     super(master)
   }
 
   static fromStyle({ name, style, document } = {}) {
+    console.warn(
+      `\`SharedStyle.fromStyle({ name, style, document })\` is deprecated. Use \`document.sharedLayersStyles.push({ name, style })\` (or \`document.sharedTextStyles\`) instead.`
```
Comment (bomberstudios) [inline Source/dom/models/SharedStyle.js]: Typo:
```suggestion
      `\`SharedStyle.fromStyle({ name, style, document })\` is deprecated. Use \`document.sharedLayerStyles.push({ name, style })\` (or \`document.sharedTextStyles\`) instead.`
```

File: Source/dom/models/SharedStyle.js (hunk 1 above)
Comment (bomberstudios) [inline Source/dom/models/SharedStyle.js]: Typo:
```suggestion
        `Cannot create a SharedStyle directly, use \`document.sharedLayerStyles.push({ name, style })\` (or \`document.sharedTextStyles\`) instead.`
```

File: docs/api/SharedStyle.md [hunk 3]
```diff
@@ -9,44 +9,37 @@ var SharedStyle = require('sketch/dom').SharedStyle
...
 
 ```javascript
-var sharedStyle = SharedStyle.fromStyle({
+document.sharedTextStyles.push({
   name: 'Header 1',
   style: text.style,
-  document: document,
+})
+
+document.sharedLayersStyles.push({
```
Comment (bomberstudios) [inline docs/api/SharedStyle.md]: Typo:
```suggestion
document.sharedLayerStyles.push({
```

**Pull request merged**